grails-plugin-url-mappings/src/main/groovy/org/grails/plugins/web/mapping/UrlMappingsAutoConfiguration.java

```diff
@@ -0,0 +1,76 @@
+package org.grails.plugins.web.mapping;
+
+import grails.config.Settings;
+import grails.util.Environment;
+import grails.web.CamelCaseUrlConverter;
+import grails.web.HyphenatedUrlConverter;
+import grails.web.UrlConverter;
+import grails.web.mapping.LinkGenerator;
+import grails.web.mapping.UrlMappings;
+import grails.web.mapping.cors.GrailsCorsConfiguration;
+import grails.web.mapping.cors.GrailsCorsFilter;
+import org.grails.web.mapping.CachingLinkGenerator;
+import org.grails.web.mapping.DefaultLinkGenerator;
+import org.grails.web.mapping.UrlMappingsHolderFactoryBean;
+import org.grails.web.mapping.mvc.UrlMappingsInfoHandlerAdapter;
+import org.grails.web.mapping.servlet.UrlMappingsErrorPageCustomizer;
+import org.springframework.beans.factory.ObjectProvider;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.AutoConfiguration;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnWebApplication;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.*;
+import org.springframework.core.type.AnnotatedTypeMetadata;
+
+@AutoConfiguration
+@ConditionalOnWebApplication(type = ConditionalOnWebApplication.Type.SERVLET)
```

Comment:
Why the choice of Servlet for the type restriction?  Is this set correctly for the shell, etc?

Reply:
how is `grails-plugin-url-mappings` used for shell? 

"`Condition that checks for the presence or absence of WebApplicationContext.`"
`Type.SERVLET` = `Only servlet-based web application will match.`. Eg. uses DispatcherServlet. Not Webflux

Reply:
There isn't any restriction under the old code and now there is.  For example, there is a value for REACTIVE; would we ever use that in grails?

Reply:
REACTIVE uses Webflux.  We require using DispatcherServlet